### DISCUSSION TITLE
Update BLazor

### DISCUSSION THREAD

**Discussion by RodrigoCardosoSantos** [balta-io/forum#556] [Tira Dúvidas] [answered]
Ola, estou fazendo o curso de blazor .net8 porem a parte de update nao funciona.
estou, fazendo sem autenticação.
reparei que mesmo alteranto o campo imputText e como se ele nao atualizase  o objeto e quando salvo ele salva a mesma informação que ja tinha cadastrada.
segue codigo abaixo.

@page "/combustivel/edit/{Id:int}"
@inject AppDbContext Context
@inject NavigationManager Navigation
@rendermode InteractiveServer

<PageTitle> Combustivel Alteração</PageTitle>
<br />
<h3 style="text-align: center;">Combustivel Alteração</h3>

@if (Model is null)
{
    <p class="text-center">
        <em>Combustivel não encontrado</em>
    </p>
}
else
{
    <h3>@Model.CH_COMBUSTIVEL </h3>
    <EditForm Model="@Model" OnValidSubmit="OnValidSubmit" FormName="CombustivelForm">
        <DataAnnotationsValidator />
        <ValidationSummary />

        <div class="mb-3">
            <label class="form-label">Descrição</label>
            <InputText @bind-Value="Model.CH_COMBUSTIVEL" class="form-control" />
            <ValidationMessage For="@(() => Model.CH_COMBUSTIVEL)" />
        </div>

        <button type="submit" class="btn btn-primary">
            Atualizar
        </button>
        <a href="/teste">Cancelar</a>
    </EditForm>

}

@code {

    [Parameter]
    public int Id { get; set; }

    public Teste? Model { get; set; }

    protected override async Task OnInitializedAsync()
    {
        Model = await Context
            .teste
            .AsNoTracking()
            .FirstOrDefaultAsync(x => x.CD_COMBUSTIVEL == Id);
    }

    public async Task OnValidSubmit()
    {
        Context.teste.Update(Model);
        await Context.SaveChangesAsync();
        Navigation.NavigateTo("combustivel");
    }
}

**Comment by andrebaltieri** [comment, marked as answer]
Boa tarde, @RodrigoCardosoSantos, como vai?
Qual erro ocorre?

**Reply by RodrigoCardosoSantos** [reply]
Nao chega apresentar nenhum erro em tela,
mais parece que o formulario nao esta funcionando, tipo quando clico em editar ele abre a tela mais nao preenche o campo 
<InputText @bind-Value="Model.CH_COMBUSTIVEL" class="form-control" />
entao eu faço um teste para ver se salva a alteração..coloco qualquer informação no ImputText..
e quando olho a Linha
 Context.Combustiveis.Update(Model);
 
 em model...Model.CH_COMBUSTIVEL esta com a informação antiga
 
 este e o codigo do Formulario
 @page "/combustivels/edit/{id:int}"
@using Services
@inject NavigationManager Navigation
@inject AppDbContext Context
@rendermode InteractiveServer


<PageTitle> Combustivel Alteração</PageTitle>
<br />
<h3>Combustivel Alteração</h3>


<h1>@Model.CD_COMBUSTIVEL</h1>
<EditForm Model="@Model" OnValidSubmit="OnValidSubmitAsync">
    <DataAnnotationsValidator />
    <ValidationSummary />

    <div class="mb-3">
        <label class="form-label">Descrição</label>
        <InputText @bind-Value="Model.CH_COMBUSTIVEL" class="form-control" />
        <ValidationMessage For="@(() => Model.CH_COMBUSTIVEL)" />
    </div>

    <button type="submit" class="btn btn-primary">
        Salvar
    </button>
    <a href="/combustivels">Cancelar</a>
</EditForm>

**Reply by andrebaltieri** [reply]
@RodrigoCardosoSantos e qual valor vem no `Model.CH_COMBUSTIVEL`?

**Reply by RodrigoCardosoSantos** [reply]
![image](https://github.com/balta-io/forum/assets/34938018/460d1ec7-5ee2-42e6-ab63-567faa93e603)
 tanto quando na funcao OnValidSubmit quanto na funcao  OnInitializedAsync quando abro o objeto model mostra assim

**Reply by andrebaltieri** [reply]
@RodrigoCardosoSantos bom dia! Esta é a tela do inspecionador do Visual Studio, eu preciso ver o erro de fato.
PS:. Cuidado para não expor nenhuma informação sensível.

**Reply by RodrigoCardosoSantos** [reply]
Opa Balta, bom dia tudo bem?
entao esse projeto é todo um teste para um outro projeto que esta em produção na empresa que trabalho.
Ele aparentemente nao me apresentou erro na parte do visual studio, so nao carregou as informações, acredito que seja algo relacionado  ao formulario. ou a versão do .net 8 porque na versao ,net 7 funciona.
uma questao  que funcinou legal foi a QuickGrid, essa opção acha que seria bom colocar em uma empresa? sera que isso poderia nao ter continuidade no futuro e causar problemas na manutenção do projeto?
Posso adicionar o projeto teste no github para dar uma olhada melhor, caso ache necessario.

**Comment by RodrigoCardosoSantos** [comment]
Consegui Resolver o problema, Por algum motivo o meu projeto estava com erro de referencia, isso estava impactando na hora de efetuar o Update.
Removi a referencia e instalei a versão certa.
